backend/src/federation/validateCommunities.ts

```diff
@@ -0,0 +1,74 @@
+import { Community as DbCommunity } from '@entity/Community'
+import { IsNull } from '@dbTools/typeorm'
+// eslint-disable-next-line camelcase
+import { requestGetPublicKey as v1_0_requestGetPublicKey } from './client/1_0/FederationClient'
+// eslint-disable-next-line camelcase
+import { requestGetPublicKey as v1_1_requestGetPublicKey } from './client/1_1/FederationClient'
+import { backendLogger as logger } from '@/server/logger'
+import { ApiVersionType } from './enum/apiVersionType'
+
+export async function startValidateCommunities(timerInterval: number): Promise<void> {
+  logger.info(
+    `Federation: startValidateCommunities loop with an interval of ${timerInterval} ms...`,
+  )
+  // better to use setTimeout twice than setInterval once -> see https://javascript.info/settimeout-setinterval
+  setTimeout(function run() {
+    validateCommunities()
+    setTimeout(run, timerInterval)
+  }, timerInterval)
+}
+
+export async function validateCommunities(): Promise<void> {
+  const dbCommunities: DbCommunity[] = await DbCommunity.createQueryBuilder()
+    .where({ verifiedAt: IsNull() })
+    .orWhere('verified_at < last_announced_at')
+    .getMany()
+  /*
+  const dbCommunities: DbCommunity[] = await DbCommunity.getRepository().manager.query(
+    'SELECT * FROM `communities` `Community` WHERE (`Community`.`verified_at` IS NULL OR `Community`.`verified_at` < `Community`.`last_announced_at`)',
+  )
+  */
+  logger.debug(`Federation: found ${dbCommunities.length} dbCommunities`)
+  if (dbCommunities) {
+    dbCommunities.forEach(async function (dbCom) {
+      logger.debug(`Federation: dbCom: ${JSON.stringify(dbCom)}`)
+      const apiValueStrings: string[] = Object.values(ApiVersionType)
+      logger.debug(`suppported ApiVersions=`, apiValueStrings)
+      if (apiValueStrings.includes(dbCom.apiVersion)) {
+        logger.debug(
+          `Federation: validate publicKey for dbCom: ${dbCom.id} with apiVersion=${dbCom.apiVersion}`,
+        )
+        const pubKey = await invokeVersionedRequestGetPublicKey(dbCom)
```

Comment:
use class based approach here:

`getAPIRequestHandler(dbCom).getPublicKey()`

Reply:
see comment above...
perhaps a part for future optimization